src/Lorentz/Contracts/RegistryDAO.hs

```diff
@@ -0,0 +1,125 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | A Registry DAO can be used to store key value pair `(k, v)`
+module Lorentz.Contracts.RegistryDAO
+  ( registryDaoContract
+  , config
+  )
+  where
+
+import Lorentz
+
+import qualified Lorentz.Contracts.BaseDAO as DAO
+import Lorentz.Contracts.RegistryDAO.Types
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+import Universum hiding (drop, swap, (>>), some)
+import Util.Markdown
+
+-- | Proposal check: `s = size(pack(proposalMetadata))` `(a * s + b)`
+registryDaoProposalCheck
+  :: forall s k v.
+     ( NicePackedValue (RegistryDaoProposalMetadata k v)
+     )
+  => DAO.ProposeParams (RegistryDaoProposalMetadata k v) : s
+  :-> Bool : s
+registryDaoProposalCheck = do
+  let a :: Natural = 1; b :: Natural = 0
+  getField #ppProposalMetadata
+  packRaw;
+  size;
+  push a; mul
+  push b; add
```

Comment:
Looks pretty beautiful, interesting to see that in stack-machine language it is possible to write such concise code :smile_cat: